src/libAtomVM/globalcontext.c

```diff
@@ -225,6 +226,13 @@ COLD_FUNC void globalcontext_destroy(GlobalContext *glb)
     }
     synclist_destroy(&glb->resource_types);
 
+    struct ListHead *select_events = synclist_nolock(&glb->select_events);
+    MUTABLE_LIST_FOR_EACH (item, tmp, select_events) {
+        struct SelectEvent *select_event = GET_LIST_ENTRY(item, struct SelectEvent, head);
+        free((void *) select_event);
```

Comment:
This is of course required, but I believe it can lead to danger if select is not stopped (via `ERL_NIF_SELECT_STOP`).  I mean, it's needed, but it can crash the VM if there is a logic error in the client that uses select.  I don't see a good way to prevent this except for testing.

Reply:
I am confused about the crashing scenario here.
Global context is only destroyed after the scheduler are done, and therefore will no longer select.

Reply:
I thought I had seen cases where if I neglected to stop a select and the VM terminated I would get a crash.  I can check, but it's not obvious.